src/components/hv-date-field/index.tsx

```diff
@@ -6,124 +6,18 @@
-import styles from './styles';
-
-/**
- * Component used to render the Cancel/Done buttons in the picker modal.
- */
-const ModalButton = (props: ModalButtonProps) => {
-  const [pressed, setPressed] = useState(false);
-
-  return (
-    <TouchableWithoutFeedback
-      onPress={props.onPress}
-      onPressIn={() => setPressed(true)}
-      onPressOut={() => setPressed(false)}
-    >
-      <View>
-        <Text style={props.getStyle(pressed)}>{props.label}</Text>
-      </View>
-    </TouchableWithoutFeedback>
-  );
-};
-
-/**
- * This text label of the field. Contains logic to decide how to format the value
- * or show the placeholder, including applying the right styles.
- */
-const FieldLabel = (props: FieldLabelProps) => {
-  const labelStyles: Array<StyleSheetType> = [props.style];
-  if (!props.value && props.placeholderTextColor) {
-    labelStyles.push({ color: props.placeholderTextColor });
-  }
-
-  const label: string = props.value
-    ? props.formatter(props.value, props.labelFormat)
-    : props.placeholder || '';
-
-  return <Text style={labelStyles}>{label}</Text>;
-};
-
-/**
- * The input field component. This is a box with text in it.
- * Tapping the box focuses the field and brings up the date picker.
- */
-const Field = (props: FieldProps) => {
-  // Styles selected based on pressed state of the field.
-  const [pressed, setPressed] = useState(false);
-
-  // Create the props (including styles) for the box of the input field.
-  const viewProps = createProps(props.element, props.stylesheets, {
-    ...props.options,
-    focused: props.focused,
-    pressed,
-    styleAttr: 'field-style',
-  });
-
-  const labelStyle: StyleSheetType = StyleSheet.flatten(
-    createStyleProp(props.element, props.stylesheets, {
-      ...props.options,
-      focused: props.focused,
-      pressed,
-      styleAttr: 'field-text-style',
-    }),
-  );
-
-  return (
-    <TouchableWithoutFeedback
-      onPress={props.onPress}
-      onPressIn={() => setPressed(true)}
-      onPressOut={() => setPressed(false)}
-    >
-      <View {...viewProps}>
-        <Contexts.DateFormatContext.Consumer>
-          {formatter => (
-            <FieldLabel
-              focused={props.focused}
-              formatter={formatter}
-              labelFormat={props.element.getAttribute('label-format')}
-              placeholder={props.element.getAttribute('placeholder')}
-              placeholderTextColor={props.element.getAttribute(
-                'placeholderTextColor',
-              )}
-              pressed={pressed}
-              style={labelStyle}
-              value={props.value}
-            />
-          )}
-        </Contexts.DateFormatContext.Consumer>
-        {props.children}
-      </View>
-    </TouchableWithoutFeedback>
-  );
-};
+import Modal from 'hyperview/src/core/components/modal';
+import type { PickerProps } from './types';
+import { Platform } from 'react-native';
+import type { Node as ReactNode } from 'react';
```

Comment:
I think this import is invalid. It should be: `import { ReactNode } from 'react'`.

Making that change seems to cause some issues with the JSX components in the file